### ISSUE TITLE
ColdFusion (CFML) - Last Line Comment Getting Removed

### ISSUE THREAD

**Issue by Jaredra** [Glavin001/atom-beautify#1446]
# Description

The results of beautification are not what I expect. It removes the final end comment tag (Which for CFML is a '--->' string). If the file does not end in a comment it does not cause issues, but when it pulls this out it results in a 500 error for the whole page. @prettydiff as I see you are pretty active here and this package is using your code for CFML. Thanks!

# Input Before Beautification

This is what the code looked like before:

```
...
</cfoutput>
</table> --->
```

# Expected Output

The beautified code should have looked like this:

```
...
</cfoutput>
</table> --->

OR

...
</cfoutput>
</table> 
--->
```

# Actual Output

The beautified code actually looked like this:

```
...
</cfoutput>
</table>

```

# Steps to Reproduce

1. Add code to Atom editor
2. Run command `Atom Beautify: Beautify Editor`
3. This beautified code does not look right!

# Checklist

I have:
- [x] Tried uninstalling and reinstalling Atom Beautify to ensure it installed properly
- [x] Reloaded (or restarted) Atom to ensure it is not a caching issue
- [x] Searched through existing Atom Beautify Issues at https://github.com/Glavin001/atom-beautify/issues
so I know this is not a duplicate issue
- [x] Filled out the Input, Expected, and Actual sections above or have edited/removed them in a way that fully describes the issue.
https://gist.github.com/Jaredra/e7de28a489bb60189914e7fcc92e7a18

**Comment by prettydiff** [comment]
Please submit the required debug.md file to a gist before we can assist you.

**Comment by Jaredra** [comment]
https://gist.github.com/Jaredra/e7de28a489bb60189914e7fcc92e7a18

Sorry about that, first time using git and wasn't aware of the protocol. I assume this is acceptable for both issues?

**Comment by prettydiff** [comment]
I am having trouble reproducing this issue at all.  Would you be able to provide a small code sample that demonstrates this problem?

**Comment by Jaredra** [comment]
Yep, just narrowed it down to nested comments. Running Beautify on this code will remove the last ending comment tag:

`

 --->`

**Comment by prettydiff** [comment]
I can reproduce the defect in the older version of Pretty Diff that is currently in Atom Beautify, but not in the latest version.  At some point I must have fixed this.... probably by accident.

Really, though, nested comments shouldn't work in practice.  So, I would recommend adjusting your code.  The parsing rules for markup stipulate that once a code comment is opened it runs until either the comment closes or the end of file is reached.  In this case the first line opens a comment and the third line closes it.  The second line is merely contents of the one comment and the fourth line is just text in HTML and a well-formedness violation in XML.

**Comment by Jaredra** [comment]
I see what you are saying and that definitely makes sense. The only pragmatic argument I can provide against that is when you are coding something, including semantic comments along the way, and then need to debug/comment out a block for whatever reason. But I can work around this one now that I'm aware of it.

Thanks again for looking into this, you sir are a godsend.

**Comment by prettydiff** [comment]
Thank you for contributing (and the kind words).
